### PR TITLE
Adding python38 into travis and tox.

### PR DESCRIPTION
**Pull request by awais786**
https://openedx.atlassian.net/browse/BOM-1569

- Removing django prior 2.2
- Adding py3.8
- Added allowd failures for 3.8 ( build is failing on 3.8)
- Fixing pylint issue.
- Update the version in setup.py and will release after merge.

ora2 shows failures on py38 so i have created a followup [ticket](https://openedx.atlassian.net/browse/BOM-1582) for fix.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,6 +21,10 @@ matrix:
     - python: 3.5
       env: TOX_ENV=quality
 
+  allow_failures:
+  - python: 3.8
+    env: TOX_ENV=django22
```
Comment (morenol) [inline .travis.yml:26]: I think that we can remove this line

Comment (awais786) [reply .travis.yml]: > I think that we can remove this line

Please review now.

File: .travis.yml [hunk 2]
```diff
@@ -9,19 +9,16 @@ sudo: false
...
+  - TOX_ENV=js
+  - TOX_ENV=quality
 
-matrix:
-  include:
-    - python: 3.5
-      env: TOX_ENV=js
-    - python: 3.5
-      env: TOX_ENV=quality
+jobs:
```
Comment (morenol) [inline .travis.yml:19]: Just for curiosity, why it was changed to jobs?

Comment (awais786) [reply .travis.yml:19]: Removed the matrix part and just using allow_failures only.

**Pull request merged**